yarn-project/aztec-nr/aztec/src/selector.nr

```diff
@@ -0,0 +1,15 @@
+use crate::utils::field_from_bytes;
+
+global SELECTOR_SIZE = 4;
+
+fn compute_selector<N>(signature: str<N>) -> Field {
+    let bytes = signature.as_bytes();
+    let hash = dep::std::hash::keccak256(bytes, bytes.len() as u32);
```

Comment:
proving a keccak scares me

Reply:
Note: this will only be executed in-circuit for non-constant signatures. For constant signatures the keccak gets simplified away:
```rust
fn main() -> pub Field {
    compute_selector("foo and bar")
}
```

```
+---------+------------------------+--------------+----------------------+
| Package | Language               | ACIR Opcodes | Backend Circuit Size |
+---------+------------------------+--------------+----------------------+
| test    | PLONKCSat { width: 3 } | 1            | 7                    |
+---------+------------------------+--------------+----------------------+
```